quixstreams/checkpointing/checkpoint.py

```diff
@@ -129,7 +129,11 @@ def commit(self):
 
         # Step 2. Flush producer to trigger all delivery callbacks and ensure that
         # all messages are produced
-        self._producer.flush()
+        logger.debug("checkpoint: flushing producer")
+        failed_count = self._producer.flush()
+        if failed_count:
```

Comment:
I'm unsure if `failed_count` is the best name for what's happening here.
`flush()` returns the number of messages left in the producer's queue, but at this point, it's unknown if they have failed or not.

I suggest renaming it to `unproduced_count` or similar